### PR TITLE
[KYUUBI #1404] Decouple zookeeper persistentNode from other modules

### PR DESCRIPTION
**Pull request by hddong**
### _Why are the changes needed?_

Decouple zookeeper persistentNode from other modules

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
Can we detail what's your next steps? And what the final interfaces looks like?

File: kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala [hunk 1]
```diff
@@ -205,7 +205,17 @@ object ServiceDiscovery extends Logging {
     }
   }
 
-  def createServiceNode(
+  def createServiceNodeAndGetPath(
```
Comment (yaooqinn) [inline kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala:208]: createAndGetServiceNode？

Comment (hddong) [reply kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala]: Done.

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
hi @hddong would you fix the conflicts？

**Comment by hddong** [comment]
> hi @hddong would you fix the conflicts？

Yep, done.

**Comment by yaooqinn** [comment]
My apology, it conflicts again :(

**Comment by hddong** [comment]
> My apology, it conflicts again :(

Haha, done.

**Comment by yaooqinn** [comment]
thanks, merged to master and v1.4

**Pull request closed without merging**